### PR TITLE
Merge from dotnet/runtime

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreRT.cs [hunk 1]
```diff
@@ -34,13 +34,6 @@ public static IntPtr OffsetOf(Type t, string fieldName)
             return new IntPtr(RuntimeAugments.InteropCallbacks.GetStructFieldOffset(t.TypeHandle, fieldName));
         }
 
-        private static object PtrToStructureHelper(IntPtr ptr, Type structureType)
-        {
-            object boxedStruct = InteropExtensions.RuntimeNewObject(structureType.TypeHandle);
```
Comment (MichalStrehovsky) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreRT.cs:39]: Oh, thanks for fixing the dataflow annotations. I didn't realize not running the ctor was a .NET Native bug when I was annotating it.

**Comment by MichalStrehovsky** [comment]
I think the pinvoke test is failing because the assembly is marked as __BlockAllReflectionAttribute. We either need to kick the BlittableClass and NonBlittableClass into a different assembly, or find a suitable pre-existing type in the framework :(. Maybe we can disable the test and file an issue for later?

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreRT.cs (hunk 1 above)
Comment (jkotas) [reply src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreRT.cs:39]: Each CoreCLR, Mono and NativeAOT had different behavior. Deduping the implementation fixed that :-)

**Comment by jkotas** [comment]
I have fixed up some of the tests to not depend on reflection and commented out the few that cannot be fixed up. Smoketests are not meant to be comprehensive tests anyway.

**Pull request merged**